File: Varios/judge.beecrowd.com/1110.py
while True:
    n = int(input())
    if n==0:
        break
    v = []
    for i in range(1,n+1):
        v.append(i)
    s = "Discarded cards: "
    f = False
    while(len(v) > 1):
        if f:
            s = s + ", "
        f = True
        s = s + ( str(v.pop(0)) )
        v.append( v.pop(0) ) 
    print(s)
    print("Remaining card: " + str(v[0]))

File: Varios/judge.beecrowd.com/1110.cxx
#include <bits/stdc++.h>
using namespace std;
int main(int argc, char **argv)
{
	queue< int > c;
	int n;
	bool f=false;
	while(cin>>n && n)
	{
		for(int k=1;k<=n;k++)
		{
			c.push(k);
		}
		cout<<"Discarded cards: ";
		f = false;
		while(c.size()>1)
		{
			if(f)
				cout<<", ";
			f = true;
			cout<<c.front();
			c.pop();
			c.push(c.front());
			c.pop();
		}
		cout<<endl;
		cout<<"Remaining card: ";
		cout<<c.front()<<endl;
		while(!c.empty())
		{
			c.pop();
		}
	}
	return 0;
}

